torch/csrc/jit/passes/constant_propagation.cpp

```diff
@@ -60,6 +76,39 @@ struct ConstantPropagator {
     return var_outputs;
   }
 
+  // Tuples are not representable as constants, however
+  // we can try to insert each tuple element and then create a TupleConstruct
+  // from the elements
+  Value* tryInsertTuple(const IValue& tuple, Value* tuple_to_replace) {
+    bool inserted_tuple = true;
+    auto type = tuple_to_replace->type();
+    TupleTypePtr tup_type;
+    if (auto opt = type->cast<OptionalType>()) {
+      tup_type = opt->getElementType()->expect<TupleType>();
+    } else {
+      tup_type = type->expect<TupleType>();
+    }
+    auto type_elements = tup_type->elements();
+    const auto& tuple_elements = tuple.toTuple()->elements();
+    std::vector<Value*> inputs;
+    for (size_t i = 0; i < type_elements.size() && inserted_tuple; ++i) {
+      auto inp =
+          tryInsertConstant(*graph_, tuple_elements[i], type_elements[i]);
+      if (inp) {
+        inputs.push_back(*inp);
+      } else {
+        inserted_tuple = false;
```

Comment:
Couldn't you `return nullptr` here and get rid of this boolean? Why do you need to keep trying to constify-elements if any one of them fails?

Reply:
True